packages/firestore/test/unit/specs/spec_builder.ts

```diff
@@ -646,6 +699,248 @@ export class SpecBuilder {
   }
 }
 
-export function spec() {
+/**
+ * SpecBuilder that supports serialized interactions between different clients.
+ *
+ * Use `client(clientIndex)` to switch between clients.
+ */
+export class MultiClientSpecBuilder extends SpecBuilder {
```

Comment:
Sorry, I never looked at this until now, but why do we need an inherited class for this?  I am not wild about needing to override every method...  that kinda' defeats the purpose of inheritance.

In an effort to avoid spinning on this PR forever, I'd be okay leaving this as-is with a TODO(multitab) to revisit (ideally in the near-term).

Reply:
I was torn on this as well (and still am). I did this because
- It makes it obvious what functionality does not need to be ported.
- The whole builder aspect makes much more sense for a single client. `clientA().build().build(`) and `clientB().build().build()` should be independent. We could do this with an improved version MultiSpecBuilder and swap out the underlying instance of SpecBuilder every time `client(...)` gets called.

That being said, I am open to remove it for now (in a follow up).